internal/supervisor/cluster.go

```diff
@@ -171,10 +171,10 @@ func (s *ClusterSupervisor) createCluster(cluster *model.Cluster, logger log.Fie
 	} else {
 		if cluster.Version != version {
 			cluster.Version = version
-			err = s.store.UpdateCluster(cluster)
-			if err != nil {
-				logger.WithError(err).Warnf("failed to set cluster version to %s", version)
-			}
+		}
+		err = s.store.UpdateCluster(cluster)
+		if err != nil {
+			logger.WithError(err).Warnf("failed to save cluster with version %s", version)
```

Comment:
No, not unless I am misunderstanding you. But there is a change in behavior here.
In the old behavior we checked to see if the k8s version had changed and wrote those changes to the Cluster representation in the database if that was the case.

Now, we write the Cluster object regardless, to persist changes to the Cluster object made by ProvisionCluster and/or CreateCluster (in this case, I need to save the actual versions of the utilities in the Cluster object to the database).

So this behavior is a little different but regarding the error string (and in the one below, your other comment), the error is only going to be the reason that the Cluster object couldn't be saved to the database for whatever reason that might be, at this time.

Reply:
Oh yeah, that is exactly what I was referring to. I just was wondering if the error should now be something more like "unable to save updated cluster" because it isn't just the version changing anymore.

Reply:
Ha! OK I see now, that's the intended meaning of the error message, but apparently it's not clear. I'll reword it to be clearer.

Reply:
Should be better now https://github.com/mattermost/mattermost-cloud/pull/143/commits/388c01c5f041013c3072e234ed46996f3b49e1e3